runelite-client/src/main/java/net/runelite/client/plugins/questlist/QuestListPlugin.java

```diff
@@ -426,4 +436,57 @@ static QuestState getByColor(int color)
 		private Widget quest;
 		private String title;
 	}
+
+	private boolean getRequirementsMet(String quest)
+	{
+		QuestRequirement questRequirements = requirements.getRequirements().get(quest);
+
+		if (questRequirements != null)
+		{
+			for (Requirement req : questRequirements.getRequirements())
+			{
+				if (!satisfiesRequirement(req))
+				{
+					return false;
+				}
+			}
+		}
+		return true;
+	}
+
+	private boolean satisfiesRequirement(Requirement r)
+	{
+		if (r instanceof OrRequirement)
```

Comment:
these should probably be methods on the classes then do a normal virtual call, look at how ItemStats works

Reply:
I don't understand what you mean by this.

If some one wants to revamp this project they can. I took it as far as I think I can for now.